engine.py

```diff
@@ -144,12 +144,20 @@ def in_plugin_mode(self):
     #####################################################################################
     # Engine Initialization and Destruction
 
-    def init_engine(self):
+    def pre_app_init(self):
         """
-        Called at Engine startup.
+        Sets up the engine into an operational state. This method called before
+        any apps are loaded.
         """
+
         self.logger.debug("%s: Initializing...", self)
 
+        local_python_path = os.path.abspath(os.path.join(self.disk_location, "python"))
+        # append the local path so that the tk_nuke and tk_nuke_qt packages can be imported
+        sys.path.append(local_python_path)
+        import tk_nuke
```

Comment:
Is it possible to use `self.import_module` here instead of having to tinker with sys.path?

Reply:
Yeah it seems import_module is built for that really, however I see this comment further down:

> \# Note! not using the import as this confuses Nuke's callback system
> \# (several of the key scene callbacks are in the main init file).

https://github.com/shotgunsoftware/tk-nuke/blob/master/engine.py#L295-L296
So perhaps there is a good reason for not doing this?

Reply:
seems like that warning should probably be added as a comment in these other locations as well?

Reply:
Actually, I tested this and it did, in fact, break it, integrations no longer loaded, and without error as well.